### PR TITLE
Feat/enable/evm

### PR DESCRIPTION
**Pull request by WrungCodes**
- EVM-compatible execution support
- `README.md` describing the purpose and structure of the Shinzohub ecosystem
- Change IBC and Cosmos SDK version

### CODE REVIEW

**Comment by dvncan** [comment]
nitpick: Cannot query for transaction TX status, please implement that route

**Comment by WrungCodes** [comment]
> nitpick: Cannot query for transaction TX status, please implement that route

This is fixed now

File: app/precompiles/viewregistry/viewregistry.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			ABI:                  newABI,
+			KvGasConfig:          storetypes.GasConfig{},
+			TransientKVGasConfig: storetypes.GasConfig{},
+		},
+		baseGas: baseGas,
+	}, nil
+}
+
+func (p Precompile) Address() common.Address {
+	return common.HexToAddress(ViewregistryPrecompileAddress)
```
Comment (dvncan) [inline app/precompiles/viewregistry/viewregistry.go:48]: Chore: Consider standardizing to ViewRegistryPrecompileAddress

Discussion: File uses ViewregistryPrecompileAddress (lowercase 'r') vs ViewRegistry in other places

File: app/precompiles/viewregistry/viewregistry.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		if err := p.AddJournalEntries(stateDB, snapshot); err != nil {
+			return nil, err
+		}
+		return bz, nil
+	})
+}
+
+func (Precompile) IsTransaction(method *abi.Method) bool {
+	switch method.Name {
+	case ViewRegistryRegisterMethod:
```
Comment (dvncan) [inline app/precompiles/viewregistry/viewregistry.go:89]: Chore: Consider standardizing to ViewRegistry*

Discussion: File uses Viewregistry* (lowercase 'r') vs ViewRegistry in other places

File: app/precompiles/viewregistry/viewregistry.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		return true
+	case ViewRegistryGetMethod:
```
Comment (dvncan) [inline app/precompiles/viewregistry/viewregistry.go:91]: Chore: Consider standardizing to ViewRegistry*

Discussion: File uses Viewregistry* (lowercase 'r') vs ViewRegistry in other places

File: app/precompiles/viewregistry/viewregistry.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+// HandleMethod handles the execution of each of the ERC-20 methods.
+func (p *Precompile) HandleMethod(
+	ctx sdk.Context,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) (bz []byte, err error) {
+	switch method.Name {
+	case ViewRegistryRegisterMethod:
```
Comment (dvncan) [inline app/precompiles/viewregistry/viewregistry.go:107]: Chore: Consider standardizing to ViewRegistry*

Discussion: File uses Viewregistry* (lowercase 'r') vs ViewRegistry in other places

File: app/precompiles/viewregistry/ViewRegistry.sol [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17;
+
+/// @dev The ViewRegistry contract's address.
```
Comment (dvncan) [inline app/precompiles/viewregistry/ViewRegistry.sol:4]: Chore: Consider standardizing to ViewRegistry*

File: proto/shinzohub/shinzohub/params.proto [hunk 6]
```diff
@@ -10,4 +10,6 @@ option go_package = "shinzohub/x/shinzohub/types";
 message Params {
   option (amino.name) = "shinzohub/x/shinzohub/Params";
   option (gogoproto.equal) = true;
-}
+
```
Comment (dvncan) [inline proto/shinzohub/shinzohub/params.proto:13]: Chore: No parameters defined

Impact: No operational configuration possible for the module.

File: cmd/shinzohubd/cmd/config.go [hunk 7]
```diff
@@ -41,12 +46,22 @@ func initAppConfig() (string, interface{}) {
...
 
 	customAppConfig := CustomAppConfig{
-		Config: *srvCfg,
+		Config:  *srvCfg,
+		EVM:     *evmserverconfig.DefaultEVMConfig(),
+		JSONRPC: *evmserverconfig.DefaultJSONRPCConfig(),
+		TLS:     *evmserverconfig.DefaultTLSConfig(),
 	}
 
+	customAppConfig.JSONRPC.Enable = true
```
Comment (dvncan) [inline cmd/shinzohubd/cmd/config.go:58]: Nitpick: consider making ENV

Will make production deployment more flexible

File: x/shinzohub/keeper/keeper.go
Comment (dvncan) [inline x/shinzohub/keeper/keeper.go:1]: **Minimal implementation**
*No business logic beyond parameter management*


Impact: Module serves no functional purpose beyond parameter storage.

Recommendation: is this still needed, can it be used in another module?

File: app/evm.go
Comment (dvncan) [inline app/evm.go:1]: **Missing EVM Test Coverage**
No ViewRegistry precompile tests
No EVM integration tests
No gas consumption tests
No error handling tests
No end-to-end transaction flow tests
Impact: Unknown behavior under edge cases and production conditions.
